src/communication/communicator.hpp

```diff
@@ -75,22 +75,26 @@ class communicator {
         gid_infos.reserve(dom_dec.num_local_cells);
 
         cell_local_size_type n_cons = 0;
+        cell_local_size_type n_gid = 0;
         std::vector<unsigned> src_domains;
         std::vector<cell_size_type> src_counts(num_domains_);
-        for (auto i: make_span(0, num_local_groups_)) {
-            const auto& group = dom_dec.groups[i];
+        for (auto g: make_span(0, num_local_groups_)) {
```

Comment:
`auto&&` but I guess you know the underlying type for make_space

Reply:
`auto` is fine here. It is going to be an integer type.